### ISSUE TITLE
Questions Regarding the Code

### ISSUE THREAD

**Issue by OopsCCK** [jyzgh/FedLPS#2]
Hello, Dr. Jia. Recently, I have read your paper and I am very grateful for the contributions you have made. However, I have some questions regarding the code associated with this paper. 

Firstly, the aggregation method proposed in the paper involves recovering the pruned model's parameters using a mask matrix before aggregation. However, I couldn't find any code related to parameter recovery. 

Secondly, the paper mentions that each device has five tasks, but in the code, it seems that each device has only one task, and there is no indication of multi-tasking. 

Thirdly, why is it that the test accuracy and loss for all devices are the same during each test?

I look forward to your response.

**Comment by YongzheJia** [comment]
Dear Chen:

Thanks for your attention to our work! 

I'm glad to address your questions:
1.  The mask matrix and the parameters used in model recovery can be found in the datastructure of "pruned_model" within our code. We utilized PyTorch's pruning tool in our FedLPS for model pruning. During the PyTorch pruning, both the mask matrix and the global model weights (aggreated in the previous round) are stored in the datastructure of "pruned_model", which are used to recover models during the FedLPS's aggregation. For more information on PyTorch pruning, this tutorial might be helpful to you: https://pytorch.org/tutorials/intermediate/pruning_tutorial.html

2. Due to the limitations of our GPU resources, we were unable to simulate 10 clients, each deployed with 5 tasks (totaling 50 tasks) in parallel. Thus, we perform mulit-task learning in a pipeline manner as the aggregation on each tasks is orthogonal. If you have access to more powerful GPUs and want to accelerate the training process, utilizing multiple threads might be beneficial.

3. All the reported test accuracies refer to the global model accuracy unless you modify our default setting (i.e., value "0") of the arguments. Please refer to explaintion of the argument "local_test" in "main_fadavg.py" file of our code for details. This work focuses on global model accurcy, but you can feel free to set "local_test" to "1" to explore personalized FL methods or extend FedLPS based on our code. In that case, the clients will report different accuracy as you will see.

I hope these answers are helpful to you. 

Once again, thank you for your interest in our work!

**Comment by OopsCCK** [comment]
Dear Jia,

Thank you very much for your response.These questions have been bothering me for a long time, and your answers have been incredibly helpful. Based on your feedback, I will gain a deeper and more detailed understanding of your research and the work you have done. Once again, I appreciate your contributions in this field.
